chia/full_node/full_node.py

```diff
@@ -948,22 +950,26 @@ async def sync_from_fork_point(
         target_peak_sb_height: uint32,
         peak_hash: bytes32,
         summaries: List[SubEpochSummary],
-    ):
+    ) -> None:
         buffer_size = 4
         self.log.info(f"Start syncing from fork point at {fork_point_height} up to {target_peak_sb_height}")
-        peers_with_peak = self.get_peers_with_peak(peak_hash)
+        peers_with_peak: List[ws.WSChiaConnection] = self.get_peers_with_peak(peak_hash)
         fork_point_height = await check_fork_next_block(
             self.blockchain, fork_point_height, peers_with_peak, node_next_block_check
         )
         batch_size = self.constants.MAX_BLOCK_COUNT_PER_REQUESTS
 
-        async def fetch_block_batches(batch_queue: asyncio.Queue, peers_with_peak: List[ws.WSChiaConnection]):
+        async def fetch_block_batches(
+            batch_queue: asyncio.Queue[Optional[Tuple[ws.WSChiaConnection, List[FullBlock]]]]
+        ) -> None:
+            start_height, end_height = 0, 0
+            new_peers_with_peak: List[ws.WSChiaConnection] = peers_with_peak[:]
             try:
                 for start_height in range(fork_point_height, target_peak_sb_height, batch_size):
                     end_height = min(target_peak_sb_height, start_height + batch_size)
                     request = RequestBlocks(uint32(start_height), uint32(end_height), True)
                     fetched = False
-                    for peer in random.sample(peers_with_peak, len(peers_with_peak)):
+                    for peer in random.sample(new_peers_with_peak, len(peers_with_peak)):
```

Comment:
In so much as the `new_peers_with_peak` is needed, probably ought to reference it's length?

```suggestion
                    for peer in random.sample(new_peers_with_peak, len(new_peers_with_peak)):
```